### ISSUE TITLE
[BUG] Compilation errors by VS Code when choosing TMC driver and MKS DGUS

### ISSUE THREAD

**Issue by McDAK33** [MarlinFirmware/Marlin#21289]
### Bug Description
* Hardware:
 __MKS Robin Nano V3 (STM32F407VGT6) + TMC2209 + MKS H43 LCD__

* Building:
__Marlin-bugfix-2.0.x (Downloaded on 08.03.2021)__

* Configuration.h: 
__#define X_DRIVER_TYPE  TMC2209
#define Y_DRIVER_TYPE  TMC2209
#define Z_DRIVER_TYPE  TMC2209
. . . 
#define DGUS_LCD_UI_MKS__

* Compilation result:
![image](https://user-images.githubusercontent.com/80269041/110349010-51e41b80-8043-11eb-8a4b-d4249342728a.png)

### Additional Information
If only TMC drivers or only MKS DGAS are defined in Configuration.h, then the assembly is compiled without errors. If this is determined at the same time, then an error occurs.

**Comment by thisiskeithb** [comment]
When reporting an issue, please be sure to follow all instructions at the top of the template, including debugging steps, and leave the sections headings intact.

**This includes attaching your configuration files, regardless of whether you think they are involved.**

<details><summary>For reference, these are the instructions included in the template:</summary>

### Before Reporting a Bug

1. Test with the `bugfix-2.0.x` branch to see whether the issue still exists.

2. Get troubleshooting help from the Marlin community to confirm it's a bug and not just a configuration error. Links at https://github.com/MarlinFirmware/Marlin/issues/new/choose

### Instructions

1. Fill out every section of the template below.

2. Always attach configuration files, regardless of whether you think they are involved.

3. Read and understand Marlin's Code of Conduct. By filing an Issue, you are expected to comply with it, including treating everyone with respect: https://github.com/MarlinFirmware/Marlin/blob/master/.github/code_of_conduct.md

---

### Bug Description

_Describe the bug in this section._

### Configuration Files

**Required:** Include a ZIP file containing `Configuration.h` and `Configuration_adv.h`.

If you've made any other modifications describe them in detail here.

### Steps to Reproduce

_Describe the steps needed to reproduce the issue._

1. [First Step]
2. [Second Step]
3. [and so on...]

**Expected behavior:**

_Describe what you expected to happen here._

**Actual behavior:**

_Describe what actually happens here._

#### Additional Information

* Provide pictures or links to videos that clearly demonstrate the issue.
* See [Contributing to Marlin](https://github.com/MarlinFirmware/Marlin/blob/2.0.x/.github/contributing.md) for additional guidelines.
</details>

**Comment by thisiskeithb** [comment]
Here's the test config I created for this issue: [Marlin-issue-21289.zip](https://github.com/MarlinFirmware/Marlin/files/6103165/Marlin-issue-21289.zip)

The Trinamic/StealthChop include in [`DGUSDisplayDef.cpp (L38-40)`](https://github.com/MarlinFirmware/Marlin/blob/aad0f517b45c03da54b75f8cac00a4a91e86f520/Marlin/src/lcd/extui/lib/dgus/mks/DGUSDisplayDef.cpp#L38-L40) needs to be patched so it's properly included:

```diff
#if ENABLED(HAS_STEALTHCHOP)
-  #include "../../../../module/stepper/trinamic.h"
+  #include "../../../../../module/stepper/trinamic.h"
#endif
```

This config will still not build properly, but the issue is because it you have to enable `SENSORLESS_HOMING`.

I'm looking into why that's a requirement.

**Comment by McDAK33** [comment]
Thank you for quick answer. I don't use SENSORLESS HOMING. I will wait for a decision.

**Comment by thisiskeithb** [comment]
PR https://github.com/MarlinFirmware/Marlin/pull/21388 fully resolves compiling issues (and fixes another bug), but I don't have the hardware to test.

You can pull those changes into your copy of Marlin for testing or wait until it's merged into `bugfix-2.0.x` and pull a fresh copy.
